File: R/day_02.R
input <- read.csv("day_2/day_2.txt", header = FALSE)

# Part 1
input <- do.call(rbind, rbind(strsplit(input$V1, "\\s+")))

df <- data.frame(
    dir = input[,1],
    n = input[,2]
)

directions <- lapply(
    split(df, df$dir),
    function(x) {
        sum(as.numeric(x$n))
        }
)

# Part 1 Answer
print((directions$down - directions$up) * directions$forward)

# Part 2
df$n <- as.numeric(df$n)

df$n[df$dir == "up"] <- df$n[df$dir == "up"] * -1

aim <- 0
depth <- 0

for (ii in seq_along(df$dir)) {
    if (df$dir[ii] == "forward") {
        depth <- depth + (aim * df$n[ii])
    } else {
        aim <- aim + (df$n[ii])
    }
}

# Part 2 Answer
print(depth * directions$forward)

File: python/day_02.py
import pandas as pd

f = open("Coding/AdventOfCode_2021_python/day_2.txt", "r")

input = f.readlines()

f.close()

def split_input(x):
  return x.split(" ", -1)

up = 0
dwn = 0
fwd = 0

for ii in input:
  tmp = split_input(ii)
  if tmp[0] == "up":
    up += int(tmp[1])
  elif tmp[0] == "down":
    dwn += int(tmp[1])
  elif tmp[0] == "forward":
    fwd += int(tmp[1])

# part 1 answer    
print((dwn - up) * fwd)

aim = 0
dpt = 0

for ii in input:
  tmp = split_input(ii)
  if tmp[0] == "forward":
    dpt += (aim * int(tmp[1]))
  if tmp[0] == "up":
    aim += int(tmp[1]) * -1
  elif tmp[0] == "down":
    aim += int(tmp[1])

# part 2 answer
print(dpt * fwd)
